modules/Cockpit/assets/components/field-layout.tag

```diff
@@ -522,7 +522,9 @@
         }
 
         getPreview(component) {
-            //console.log(component)
+            var triggerOutput = [];
+            App.trigger('field.layout.component.preview', {component: component, def: this.components[component.component], output: triggerOutput});
+            if (triggerOutput.length > 0) return triggerOutput.join('');
```

Comment:
Why an array?

Reply:
For the handler to be able to send back data it has to be some kind of referenceable data type, ie. object or array. A string would not be able to be changed from inside the handler.

Here is some code of the handler in our addon that uses this trigger:
```
 App.on("field.layout.component.preview", function (trigger) {
        var componentName = trigger.params.component.component;
        var value = trigger.params.component.settings;
        var output = trigger.params.output;
        switch (componentName) {
            case "custom_heading":
                output.push(badge(value.tag) + value.text);
                return;
```

Reply:
@aheinze Any toughts on this? I just rebased the PR to stay in sync.

Reply:
```js
var output = '',
    params = {
        component: component, 
        def: this.components[component.component],
        render: function(contents) {
            output = contents || '';
        }
    };

App.trigger('field.layout.component.preview', params);

if (output) return output;
```

your handler would look then like:

```js
 App.on("field.layout.component.preview", function (trigger) {
        var componentName = trigger.params.component.component;
        var value = trigger.params.component.settings;
        var render = trigger.params.render;
        switch (componentName) {
            case "custom_heading":
                render(badge(value.tag) + value.text);
                return;
```

Reply:
Great, I will rewrite it!